tools/RAiDER/utilFcns.py

```diff
@@ -1222,3 +1223,36 @@ def get_dt(t1,t2):
      18000.0
     '''
     return np.abs((t1 - t2).total_seconds())
+
+
+def calc_buffer(model:str, lat:float, lon:float):
+    """ Calculate the buffer for ray tracing from the latitude and model
+
+    Quick and dirty estimate
+    """
+    from shapely.geometry import Point
+    from shapely.ops import transform
+    if -90 < lat < 90:
+        proj = False
+    else:
+        proj = True
+        lon, lat = transform_coords(4087, 4326, lon, lat)
+
+    # get the top of the model
+    dct_tom = {'HRES': 80301.65, 'HRRR': 80301.65, 'GMAO': 80301.65, 'ERA5': 80301.65}
+    tom     = dct_tom[model]
+    near_range = tom / np.sin(np.deg2rad(30))
+
+    # this is in meters tho
+    buffer   = near_range / np.cos(np.deg2rad(lat))
+
+    x, y = transform_coords(4326, 4087, lon, lat)
+    poly = Point(x, y).buffer(buffer)
```

Comment:
based on conversation what the lon lat is the centroid of the ROI. You are now using the centroid and adding buffer. So you missed now half of the ROI.

Reply:
The buffer you need to add to the extends of the ROI

Reply:
So i think you need to have the BOX and then transform it to a local coordinate system in m; then path it with the buffer; and covert it back to the source proj; that would be the new ROI.